src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs

```diff
@@ -22,9 +23,10 @@ public static class MongoDbHealthCheckBuilderExtensions
         /// <returns>The <see cref="IHealthChecksBuilder"/>.</returns></param>
         public static IHealthChecksBuilder AddMongoDb(this IHealthChecksBuilder builder, string mongodbConnectionString, string name = default, HealthStatus? failureStatus = default, IEnumerable<string> tags = default)
         {
+            var mongoDbHealthCheck = new MongoDbHealthCheck(mongodbConnectionString);
             return builder.Add(new HealthCheckRegistration(
                 name ?? NAME,
-                sp => new MongoDbHealthCheck(mongodbConnectionString),
+                sp => mongoDbHealthCheck,
                 failureStatus,
                 tags));
         }
```

Comment:
I wonder why you want to create MongoDbHealthCheck object every time. In it you have MongoClient which should be reused (http://mongodb.github.io/mongo-csharp-driver/2.0/reference/driver/connecting/#re-use)

Reply:
You can check ServiceBusHealthCheck in order to check how solve similar issues

Reply:
Could you provide the link? Can't find.

Reply:
The idea is do something similar to 
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.AzureServiceBus/AzureServiceBusTopicHealthCheck.cs to preserve static MongoClient but enable different AddMongoDb health registrations